tasks/setup_environment.yml

```diff
@@ -2,7 +2,8 @@
 - name: Set project and service identifier in facts
   ansible.builtin.set_fact:
     project: "{{ group_names[0] }}"
-    service_identifier: "{{ group_names[0] }}{{ '' if type in default_node_types else '-' + type }}"
+    service_identifier: >-
+      {{ group_names[0] }}{{ '-' + client if type in default_node_types and client is defined else '-' + type + ('-' + client if client is defined else '') }}
```

Comment:
Looks too complex, lets define client suffix seprately, you can adapt something similar to
```
client_suffix: "{{ ('-' + client) if client is defined else '' }}"
```

and then we can use it
```
service_identifier: "{{ group_names[0] }}{{ '' if type in default_node_types else '-' + type }}{{ client_suffix }}"
```